### PR TITLE
CI: Add CI Support

### LINKED ISSUE #2: 使用 Continuous Integration 工具进行持续集成
如题

### PR DESCRIPTION
**Pull request by gaocegege**
Close #2

### CODE REVIEW

**Comment by gaocegege** [comment]
travis 最高 ubuntu 14.04, 好蠢

**Comment by at15** [comment]
我记得还有个 windows 环境的 CI 吧? https://www.appveyor.com/ (只是可能每次都需要装个 texlive 很凶残啊 ...)

**Comment by gaocegege** [comment]
嗯，现在的问题是没有 ctex 支持

**Comment by gaocegege** [comment]
日狗, 放弃了...

**Comment by AlexaraWu** [comment]
不要用 CI 自带的包管理器安装 TeX Live，直接从 CTAN 安装最新的 TeX Live 就行。

参见上游我提交的的 PR [weijianwen/SJTUThesis#222](https://github.com/weijianwen/SJTUThesis/pull/222)。

**Comment by gaocegege** [comment]
Thanks for your info @AlexaraWu

**Comment by gaocegege** [comment]
@at15 Rebase needed

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - export "PATH=$TEXBIN:$PATH"
+
+  - sudo $TEXBIN/tlmgr install latexmk ulem
+
+before_script:
+  - sed -i 's/,\s*fontset=\?\w*//' thesis.tex
+  - sed -i '/setmainfont/d' sjtuthesis.cls
+
+script:
+  - make validate
```
Comment (gaocegege) [inline .travis.yml]: New line, plz.

Comment (at15) [reply .travis.yml]: maybe add `.editorconfig`  ?

**Comment by at15** [comment]
@gaocegege it seems your are not using Makefile and use `xelatex` directly in your previous commit, so just stick with `make validate` or try with your old commands? I am not an expert in typeset.

**Comment by gaocegege** [comment]
@at15 `make validate` works for me :-)

**Comment by gaocegege** [comment]
Sorry I have no time on this repo since I have a project with caicloud and it takes me much time. @at15 Do what you want and go ahead :-)

**Comment by gaocegege** [comment]
@AlexaraWu Thanks for your contribution. 你愿意成为 dyweb/SJTUThesis 的 outside collaborator 么 :-)

**Comment by AlexaraWu** [comment]
@gaocegege With pleasure! Thank you for your invitation.

**Comment by at15** [comment]
@AlexaraWu pls check https://github.com/dyweb/SJTUThesis/invitations

**Pull request merged**